x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/template/IndexTemplateRegistryTests.java

```diff
@@ -0,0 +1,403 @@
+        assertThat(processors, hasSize(1));
+        Map<?, ?> setProcessor = (Map<?, ?>) ((Map<?, ?>) processors.get(0)).get("set");
+        assertNotNull(setProcessor.get("field"));
+        assertNotNull(setProcessor.get("copy_from"));
+        assertNotNull(listener);
+        calledTimes.incrementAndGet();
+    }
+
+    private ClusterChangedEvent createClusterChangedEvent(Map<String, Integer> existingTemplates, DiscoveryNodes nodes) {
+        return createClusterChangedEvent(existingTemplates, Collections.emptyMap(), Collections.emptyMap(), nodes);
+    }
+
+    private ClusterChangedEvent createClusterChangedEvent(
+        Map<String, Integer> existingTemplates,
+        Map<String, LifecyclePolicy> existingPolicies,
+        Map<String, Integer> existingIngestPipelines,
+        DiscoveryNodes nodes
+    ) {
+        ClusterState cs = createClusterState(Settings.EMPTY, existingTemplates, existingPolicies, existingIngestPipelines, nodes);
+        ClusterChangedEvent realEvent = new ClusterChangedEvent(
+            "created-from-test",
+            cs,
+            ClusterState.builder(new ClusterName("test")).build()
+        );
+        ClusterChangedEvent event = spy(realEvent);
+        when(event.localNodeMaster()).thenReturn(nodes.isLocalNodeElectedMaster());
+
+        return event;
+    }
+
+    private ClusterState createClusterState(
+        Settings nodeSettings,
+        Map<String, Integer> existingComponentTemplates,
+        Map<String, LifecyclePolicy> existingPolicies,
+        Map<String, Integer> existingIngestPipelines,
+        DiscoveryNodes nodes
+    ) {
+        Map<String, ComponentTemplate> componentTemplates = new HashMap<>();
+        for (Map.Entry<String, Integer> template : existingComponentTemplates.entrySet()) {
+            ComponentTemplate mockTemplate = mock(ComponentTemplate.class);
+            when(mockTemplate.version()).thenReturn(template.getValue() == null ? null : (long) template.getValue());
+            componentTemplates.put(template.getKey(), mockTemplate);
+        }
+
+        Map<String, LifecyclePolicyMetadata> existingILMMeta = existingPolicies.entrySet()
+            .stream()
+            .collect(Collectors.toMap(Map.Entry::getKey, e -> new LifecyclePolicyMetadata(e.getValue(), Collections.emptyMap(), 1, 1)));
+        IndexLifecycleMetadata ilmMeta = new IndexLifecycleMetadata(existingILMMeta, OperationMode.RUNNING);
+
+        Map<String, PipelineConfiguration> ingestPipelines = new HashMap<>();
+        for (Map.Entry<String, Integer> pipelineEntry : existingIngestPipelines.entrySet()) {
+            // we cannot mock PipelineConfiguration as it is a final class
+            ingestPipelines.put(
+                pipelineEntry.getKey(),
+                new PipelineConfiguration(
+                    pipelineEntry.getKey(),
+                    new BytesArray(String.format("{\"version\": %d}", pipelineEntry.getValue())),
```

Comment:
This (not the other comment) is tripping up forbidden APIs check in CI - Strings.format should be good enough instead

Reply:
Thanks!

Reply:
Yes, forbidden API's extend to test classes as well